enigbe/PrinterConnect
# account confirmation resource

### What does this PR do?
- add a timed confirmation resource that allow new clients confirm
their account and access protected endpoints
- give new clients a 30-minute time frame from creation to confirmation
of account
- allow clients request a new confirmation link sent to their email
if the initial one expires
- use UUID to codify client accounts instead of their usernames

#### Description of Task to be completed?

1. Confirm account from link sent to email: `GET '/client/confirmation/<string:confirmation_id>'`
2. Resend confirmation link to email address: `POST '/client/resend_confirmation/<string:email>'`
3. Get a list of confirmations belonging to each client: `GET '/client/resend_confirmation/<string:email>'`. **_Note_**: Primarily for test purposes

#### How should this be manually tested?

1. Create a new client: `POST /client/signup/email` with post body `{'email': 'janedoe@gmail.com',
                    'username': 'jane_d',
                    'first_name': 'jane',
                    'last_name': 'doe',
                    'password': '0123456789'}`
2. Confirm client with generated link sent to email `GET '/client/confirmation/<string:confirmation_id>'`. Replace `<string:confirmation_id>` with UUID hex in the email sent.
3. Resend confirmation (after 30 minutes of 2 above): `GET '/client/resend_confirmation/janedoe@gmail.com'`
4. View all confirmations mapped to client `GET '/client/resend_confirmation/janedoe@gmail.com'`

**_Note_**: Only authorized mailgun recipients can conduct this test

#### Any background context you want to provide?
The current client verification model does not factor link expiration and the possibility of clients re-sending confirmation emails.
It also relies on the client username in the url. By codifying the link with a UUID hex, the confirmation model is much safer. This is so because, when compared with usernames, UUID hexes are much difficult to guess.
